src/tools/getVariables.ts

```diff
@@ -0,0 +1,263 @@
+        );
+    }
+
+    // Check if project has git persistence
+    const hasGitVariables = hasVariablesInGit(project.PersistenceSettings);
+
+    if (hasGitVariables && !gitRef) {
+        throw new Error(`Missing gitRef for config-as-code project ${project.Name}`);
+    }
+
+    let resource: VariableSetResource;
+
+    if (hasGitVariables) {
+        // For git projects, we need to get both text and sensitive variables separately
+
+        // Retrieve the variable set stored in git for the associated gitRef
+        const textVariableSet = await apiClient.get<VariableSetResource>(
+            `/api/spaces/${spaceId}/projects/${project.Id}/${gitRef}/variables`
+        );
+
+        // Sensitive variables are still stored in the database so that they can be encrypted
+        const sensitiveVariableSet = await apiClient.get<VariableSetResource>(
+            `/api/spaces/${spaceId}/projects/${project.Id}/variables`
+        );
+
+        // Combine variables from both sets
+        resource = {
+            ...textVariableSet,
+            Variables: [...textVariableSet.Variables, ...sensitiveVariableSet.Variables]
+        };
+    } else {
+        // For database projects, get variables directly
+        resource = await apiClient.get<VariableSetResource>(`/api/spaces/${spaceId}/variables/${project.VariableSetId}`);
+    }
+
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    const { ScopeValues, ...result } = resource;
+
+    return result;
+}
+
+async function loadLibraryVariableSetVariables(
+    includedLibraryVariableSetIds: string[],
+    apiClient: Client,
+    spaceId: string
+): Promise<LibraryVariableSetWithVariables[]> {
+
+    if (includedLibraryVariableSetIds.length === 0) return [];
+
+    // Get library variable sets
+    const libraryVariableSets = await apiClient.get<ResourceCollection<LibraryVariableSetResource>>(
+        `/api/spaces/${spaceId}/libraryvariablesets?ids=${includedLibraryVariableSetIds.join(',')}`
+    );
+
+    // Get all variable sets for the library variable sets
+    const variableSetIds = libraryVariableSets.Items.map(lvs => lvs.VariableSetId);
+    const allVariableSets = await apiClient.get<VariableSetResource[]>(
+        `/api/spaces/${spaceId}/variables/all?ids=${variableSetIds.join(',')}`
+    );
+
+    const responseVariableSets: VariableSetResponse[] = allVariableSets.map(resource => {
+        // eslint-disable-next-line @typescript-eslint/no-unused-vars
+        const { ScopeValues, ...result } = resource;
+        return result;
+    })
+    // Create lookup map
+    const allVariableSetsMap = responseVariableSets.reduce((acc: ResourcesById<VariableSetResponse>, resource) => {
+        acc[resource.Id] = resource;
+        return acc;
+    }, {});
+
+    // Combine library variable sets with their variable sets
+    return libraryVariableSets.Items.map(lvs => ({
+        variableSet: allVariableSetsMap[lvs.VariableSetId],
+        libraryVariableSet: lvs
```

Comment:
Should we add an explanation for the model as to what variable sets and library variable sets are and their relationship with each other or is it already well understood from the response structure?

Reply:
I'm not sure if the model really needs to know about that, unless we want it to be able to modify them afterwards. Its more about evaluating the variables available to a project.